nekoyume/battle/components/behaviors/skills/__init__.py

```diff
@@ -70,120 +67,123 @@ def tick(self, simulator):
         else:
             atk += my_stats.calc_melee_atk()
         atk = math.floor(atk * (self.data.power * 0.01))
-        # find target
+        return atk
+
+    def is_cooltime(self, simulator):
+        return self.nexttime > simulator.time
+
+    def kill(self, simulator, target):
+        my_stats = self.owner.get_component(Stats)
+        target_stats = target.get_component(Stats)
+        simulator.logger.log(Dead(id_=target.id_))
+        if self.owner.type_ is CharacterType.PLAYER:
+            my_stats.get_exp(target_stats.data.reward_exp)
+            simulator.logger.log(GetExp(
+                id_=self.owner.id_,
+                exp=target_stats.data.reward_exp
+            ))
+
+    def casting(self, simulator):
+        if self.cast_remains > 0:
+            self.cast_remains -= 1
+            simulator.logger.log(Casting(
+                id_=self.owner.id_,
+                tick_remain=self.cast_remains,
+            ))
+            return True
+        if not self.is_casting and self.data.cast_time > 0:
+            self.is_casting = True
+            self.cast_remains = self.data.cast_time
+            simulator.logger.log(Casting(
+                id_=self.owner.id_,
+                tick_remain=self.cast_remains,
+            ))
+            return True
+        self.is_casting = False
+        return False
+
+
+# weapon based attack
+class Attack(Skill):
```

Comment:
It would be better if that comment turns to a docstring, e.g.:

```python
class Attack(Skill):
    """Weapon-based attack."""

```

Reply:
👍